### ISSUE TITLE
No screenshots

### ISSUE THREAD

**Issue by pskmsc** [cuckoosandbox/cuckoo#1561]
Hello,

I rarely have screenshots in reports.
I had some, let say in 3% of my tests.
The configuration and logs are the same as: https://github.com/cuckoosandbox/cuckoo/issues/1557
200 G free on the disc, no permissions issues.

I've read the 5 pages in the documentation where the word screenshot is appearing (doom doom wah) and I got the latest version of cuckoo.

I'm stuck on this one.

Regards,

**Comment by doomedraven** [comment]
but still no logs from `cuckoo -d` :D

**Comment by pskmsc** [comment]
RTFT/RTFL : "The configuration and logs are the same as: #1557"
Which point to https://pastebin.com/tqeqS4Cu

Complementary informations (from an analysis.log) : 

2017-05-16 13:17:56,000 [analyzer] DEBUG: Starting analyzer from: C:\tmpkdxg95
2017-05-16 13:17:56,000 [analyzer] DEBUG: Pipe server name: \??\PIPE\nryYrIPPwcmVqbLY
2017-05-16 13:17:56,015 [analyzer] DEBUG: Log pipe server name: \??\PIPE\bRvkYCvRNUFvYkLF
2017-05-16 13:17:56,296 [analyzer] DEBUG: Started auxiliary module DbgView
2017-05-16 13:17:56,765 [analyzer] DEBUG: Started auxiliary module Disguise
2017-05-16 13:17:57,203 [analyzer] DEBUG: Loaded monitor into process with pid 464
2017-05-16 13:17:57,203 [analyzer] DEBUG: Started auxiliary module DumpTLSMasterSecrets
2017-05-16 13:17:57,203 [analyzer] DEBUG: Started auxiliary module Human
2017-05-16 13:17:57,203 [analyzer] DEBUG: Started auxiliary module InstallCertificate
2017-05-16 13:17:57,203 [analyzer] DEBUG: Started auxiliary module Reboot
2017-05-16 13:17:57,280 [analyzer] DEBUG: Started auxiliary module RecentFiles
2017-05-16 13:17:57,296 [analyzer] DEBUG: Started auxiliary module Screenshots
2017-05-16 13:17:57,296 [analyzer] DEBUG: Started auxiliary module LoadZer0m0n
2017-05-16 13:17:58,000 [lib.api.process] INFO: Successfully executed process from path u'C:\\Users\\user\\AppData\\Local\\Temp\\xmd.exe' with arguments '' and pid 2960
2017-05-16 13:17:58,296 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
2017-05-16 13:17:58,375 [analyzer] DEBUG: Loaded monitor into process with pid 2960
2017-05-16 13:17:58,375 [analyzer] DEBUG: Received request to inject pid=2960, but we are already injected there.
2017-05-16 13:17:58,483 [analyzer] INFO: Added new file to list with pid 2960 and path C:\Users\user\AppData\Local\Temp\DDFC.tmp\xmd.bat
2017-05-16 13:17:58,530 [analyzer] INFO: Injected into process with pid 2124 and name u'\u4ca0\u01f6'
2017-05-16 13:17:58,696 [analyzer] DEBUG: Loaded monitor into process with pid 2124
2017-05-16 13:17:59,315 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
2017-05-16 13:18:00,321 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
2017-05-16 13:18:01,323 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed


Regards

**Comment by doomedraven** [comment]
lol my bad xD escaped between so much info, but still also will ask for provide log for only 1 analysis, as there tons and a lot of missed parts

**Comment by pskmsc** [comment]
Yep, as I have said in my ticket, sometime screenshots are working but it's very very rare.

Name: Pillow
Version: 3.1.2
Summary: Python Imaging Library (Fork)
Home-page: http://python-pillow.github.io/
Author: Alex Clark (Fork Author)
Author-email: aclark@aclark.net
License: Standard PIL License
Location: /usr/lib/python2.7/dist-packages

**Comment by doomedraven** [comment]
i can't find the cuckoo version which you have, as there was problem with screens in old version

**Comment by ixalle** [comment]
Cuckoo (2.0.3a3)

**Comment by doomedraven** [comment]
update just in case Cuckoo (2.0.3a5) as latest is this one

**Comment by ixalle** [comment]
done but same issue :( 
 
reports/report.json:771935:         "2017-05-16 15:01:25,355 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed\n",
reports/report.json:771940:         "2017-05-16 15:01:26,194 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed\n"

**Comment by pskmsc** [comment]
Adding more memory does not change anything (that was a test).
BUT, strange thing, screenshots are present if we do a RDP on the VM during the run.

**Comment by pskmsc** [comment]
I can confirm, as soon as we disconnect the RDP session, error messages about screenshot grabbing are appearing in the log:
2017-05-16 16:29:19,812 [modules.auxiliary.screenshots] ERROR:Cannot take screenshot: screen grab failed
2017-05-16 16:29:20,812 [modules.auxiliary.screenshots] ERROR:Cannot take screenshot: screen grab failed
2017-05-16 16:29:21,812 [modules.auxiliary.screenshots] ERROR:Cannot take screenshot: screen grab failed

**Comment by pskmsc** [comment]
I can confirm, as soon as we disconnect the RDP session, error messages about screenshot grabbing are appearing in the log:
2017-05-16 16:29:19,812 [modules.auxiliary.screenshots] ERROR:Cannot take screenshot: screen grab failed
2017-05-16 16:29:20,812 [modules.auxiliary.screenshots] ERROR:Cannot take screenshot: screen grab failed
2017-05-16 16:29:21,812 [modules.auxiliary.screenshots] ERROR:Cannot take screenshot: screen grab failed

**Comment by pskmsc** [comment]
I can't find the information anywhere : is a running X on the cuckoo server is mandatory to have screenshots ?

**Comment by doomedraven** [comment]
no, @jbremer any idea?

**Comment by pskmsc** [comment]
Hello,

I tried everything I can, still no screenshots, excepted when I'm connected in RDP during the capture.
As you can see here, error messages about screenshots are going away as soon as I connect in RDP (analyze logs on the left screen, cuckoo logs on the right screen):
http://imgur.com/a/cpJwU

I've made a quick test in python, Pillow seems to work properly on the VM:
http://imgur.com/a/XR6ZU

Any help is welcome.

**Comment by SparkyNZL** [comment]
what is the Virtual machinery you are using  ? Virtual box ? qemu ? Also have you tried install PIL in the VM and pillow in the host ? , is the VM windows ? 

Cheers.

**Comment by pskmsc** [comment]
Hello,

A little bump, as we are still stuck on the screenshots issue.

Regards,

**Comment by ixalle** [comment]
Hi,

we use Virtualbox, only Windows 7 VM and yes PIL and Pillow are installed on each side.

**Comment by pskmsc** [comment]
Oh sorry, browser cache issue, I didn't seen @SparkyNZL question.
We are using VirtualBox 5.1.
Pillow is installed on the the host:
Name: Pillow
Version: 3.1.2

Pillow as been installed on the VM and I made a test script, which was doing a screen shot and this worked fine.

**Comment by pskmsc** [comment]
Hello,

Little bump for this topic.

No ideas ?

Regards,

**Comment by jbremer** [comment]
Sorry for the delay, I haven't seen the following error yet:
` 2017-05-16 13:17:59,315 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed`
The following results are the first I find when googling the error.. they do make sense - can you tell us more about your setup?
* https://github.com/python-pillow/Pillow/issues/1601
* https://stackoverflow.com/questions/24370228/python-imaging-library-fails-to-grab-whole-screen

**Comment by pskmsc** [comment]
Many thanks for your answer.

Sadly, this is not helping, I still have to test the "Disable display scaling on high DPI settings" trick, but I have not time for this today.
There is no issue with scereenshots and PIL, if I'm running the VM and do a screenshot using piloow it's working perfectly.
Also, if I'm doing a RDP on the VM during the analysis, I will have screenshots.

Regards,

**Comment by jbremer** [comment]
So your VM is not running a screensaver?

**Comment by pskmsc** [comment]
No

**Comment by pskmsc** [comment]
Hello,

We tested the "Disable display scaling on high DPI settings" trick without success

Regards,

**Comment by pskmsc** [comment]
Hello,

I've tried to search on internet again and found no solutions.
Nobody have an idea we do not have tested in this thread ?

Regards,
